CHANGES.rst

```diff
@@ -19,101 +19,158 @@ backported to 0.29.x and are not strictly speaking "new" in Cython 3.0.0.
+  to match the signature in Py3.  It still has an automatic fallback for Py2.
+  (Github issue :issue:`3909`)
 
 * ``PyMem_[Raw]Calloc()`` was added to the ``cpython.mem`` declarations.
   Note that the ``Raw`` versions are no longer #defined by Cython.  The previous
   macros were not considered safe.
   Patch by William Schwartz and David Woods.  (Github issue :issue:`3047`)
 
-* Unicode module names and imports are supported.
-  Patch by David Woods.  (Github issue :issue:`3119`)
+* The runtime size check for imported ``PyVarObject`` types was improved
+  to reduce false positives and adapt to Python 3.11.
+  Patch by David Woods.  (Github issues :issue:`4827`, :issue:`4894`)
 
-* ``PyEval_InitThreads()`` is no longer used in Py3.7+ where it is a no-op.
+* The generated C code failed to compile in CPython 3.11a4 and later.
+  (Github issue :issue:`4500`)
 
-* The ``Tempita`` module no longer contains HTML processing capabilities, which
-  were found to be broken in Python 3.8 and later.
-  Patch by Marcel Stimberg.  (Github issue :issue:`3309`)
+* ``pyximport`` no longer uses the deprecated ``imp`` module.
+  Patch by Matúš Valo.  (Github issue :issue:`4560`)
 
-* Cython avoids raising ``StopIteration`` in ``__next__`` methods when possible.
-  Patch by David Woods.  (Github issue :issue:`3447`)
+* Improvements to ``PyTypeObject`` definitions in pxd wrapping of libpython.
+  Patch by John Kirkham. (Github issue :issue:`4699`)
 
-* ``__del__(self)`` on extension types now maps to ``tp_finalize`` in Python 3.
-  Original patch by ax487.  (Github issue :issue:`3612`)
+* Some old usages of the deprecated Python ``imp`` module were replaced with ``importlib``.
+  Patch by Matúš Valo.  (Github issue :issue:`4640`)
+
+* ``cpdef`` enums no longer use ``OrderedDict`` but ``dict`` in Python 3.6 and later.
+  Patch by GalaxySnail.  (Github issue :issue:`5180`)
+
+* Several problems with CPython 3.12 were resolved.
+  (Github issue :issue:`5238`)
+
+* The exception handling code was adapted to CPython 3.12.
+  (Github issue :issue:`5442`)
+
+* The Python ``int`` handling code was adapted to make use of the new ``PyLong``
+  internals in CPython 3.12.
+  (Github issue :issue:`5353`)
+  
+* A compile error when using ``__debug__`` was resolved.
+  
+* The deprecated ``_PyGC_FINALIZED()`` C-API macro is no longer used.
+  Patch by Thomas Caswell and Matúš Valo.  (Github issue :issue:`5481`)
+  
+* A crash in Python 2.7 was fixed when cleaning up extension type instances
+  at program end.
+
+
+
+
+
+Compatibility with CPython and the Python C API
+-----------------------------------------------
+
+Since Cython 3.0.0 started development, CPython 3.8-3.11 were released.
+All these are supported in Cython, including experimental support for the
+in-development CPython 3.12. On the other end of the spectrum, support for
+Python 2.6 was dropped.
+
+Cython interacts very closely with the C-API of Python, which is where most
+of the adaptation work happens.
```

Comment:
This section appears duplicated with the above. Haven't checked all items, but at least the title, intro and first few are the same.